p2p/libp2p/netMessenger.go

```diff
@@ -1308,19 +1309,37 @@ func (netMes *networkMessenger) SetPeerShardResolver(peerShardResolver p2p.PeerS
 	return nil
 }
 
-// SetCurrentBytesProvider sets the current peer bytes provider that is able to prepare the bytes to be sent to a new peer
-func (netMes *networkMessenger) SetCurrentBytesProvider(currentBytesProvider p2p.CurrentPeerBytesProvider) error {
-	if check.IfNil(currentBytesProvider) {
-		return p2p.ErrNilCurrentPeerBytesProvider
+// SetCurrentPayloadProvider sets the current payload provider that is able to prepare the bytes to be sent to a new peer
+func (netMes *networkMessenger) SetCurrentPayloadProvider(currentPayloadProvider p2p.CurrentPayloadProvider) error {
+	if check.IfNil(currentPayloadProvider) {
+		return p2p.ErrNilCurrentPayloadProvider
 	}
 
 	netMes.mutCurrentBytesProvider.Lock()
-	netMes.currentBytesProvider = currentBytesProvider
+	netMes.currentPayloadProvider = currentPayloadProvider
+	buff, isValid := currentPayloadProvider.BytesToSendToNewPeers()
 	netMes.mutCurrentBytesProvider.Unlock()
 
+	netMes.notifyExistingPeers(buff, isValid)
+
 	return nil
 }
 
+func (netMes *networkMessenger) notifyExistingPeers(buff []byte, isValid bool) {
+	if !isValid {
+		return
+	}
+
+	pids := netMes.ConnectedPeers()
+	for i := 0; i < len(pids); i++ {
+		pid := pids[i]
+		errNotCritical := netMes.SendToConnectedPeer(common.ConnectionTopic, buff, pid)
+		if errNotCritical != nil {
+			log.Trace("networkMessenger.PeerConnected", "pid", pid.Pretty(), "error", errNotCritical)
```

Comment:
`networkMessenger.SendToConnectedPeer` instead of `networkMessenger.PeerConnected` ?